nixos/modules/services/hardware/ratbagd.nix

```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ratbagd;
+in
+
+{
+
+  ###### interface
+
+  options = {
+    services.ratbagd = {
+      enable = mkOption {
+        default = false;
+        description = ''
+          Whether to enable ratbagd for configuring gaming mice.
+        '';
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    services.dbus.packages = [ pkgs.libratbag ];
+
+    # Give users access to the "ratbagctl" tool
+    environment.systemPackages = [ pkgs.libratbag ];
+
+    systemd.packages = [ pkgs.libratbag ];
+    systemd.services.ratbagd = {
```

Comment:
Why did you leave this? It should already be specified in the service file in the package, which is installed by the `systemd.packages` option.

Reply:
I couldn't figure out of that actually used the packages own unit file. It didn't seem to work for me when I removed my own definition. Is there a way to point it to where the build outputs it?

Reply:
It does, only the servcie file does not have `wantedBy`. You can do `systemd.services.ratbagd.wantedBy = [ "multi-user.target" ];`, which will add the key to the [override](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#id-1.14.3).

I wonder why the upstream service does not have that, though – maybe the daemon is supposed to be launched using a different method.